nodebuilder/share/constructors.go

```diff
@@ -52,3 +55,24 @@ func newModule(lc fx.Lifecycle, bServ blockservice.BlockService, avail share.Ava
 	})
 	return &module{serv}
 }
+
+func newEdsSub(lc fx.Lifecycle, h host.Host, network modp2p.Network) (*eds.PubSub, error) {
+	pubsub, err := eds.NewPubSub(
+		fxutil.WithLifecycle(context.Background(), lc),
+		h,
+		string(network),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	lc.Append(fx.Hook{
```

Comment:
[optional] Can you try to do this without the hooks? So no separate constructor. But just see if you like it better

Reply:
How can we stop pubsub without hooks?

Reply:
i will checkout and try it, then show you the diff in sync